### PR TITLE
Use `ViewerImportUrl` for updating history, adds more unit tests

### PR DESCRIPTION
**Pull request by Wumpf**
_Mostly_ a refactor that got split out of other ongoing work

* integrate valid values of `LogDataSource` into `ViewerImportUrl` to avoid handling the same weird corner cases all the time
* `display_mode_to_content_url` is now integrated/split into `ViewerImportUrl` via `ViewerImportUrl::from_display_mode` and `ViewerImportUrl::sharable_url`
	* `sharable_url` already handles appending the a web viewer base to it which will be relevant in the next PR

But also:
* tons of new tests for these
* fix some corner cases with the example/welcome page where it was possible to get into a state where we add an invalid server & show the welcome page
* fix some formatting issue on intra recording urls

### CODE REVIEW

File: crates/viewer/re_viewer/src/open_url.rs [hunk 1]
```diff
@@ -121,14 +161,214 @@ fn parse_webviewer_url(url: &str) -> anyhow::Result<ViewerImportUrl> {
...
+    pub fn from_display_mode(
+        store_hub: &StoreHub,
+        display_mode: DisplayMode,
+    ) -> anyhow::Result<Self> {
+        match display_mode {
+            DisplayMode::Settings => {
+                // Not much point in updating address for the settings screen.
+                Err(anyhow::anyhow!("Can't share links to the settings screen."))
+            }
+            DisplayMode::LocalRecordings => {
```
Comment (abey79) [inline crates/viewer/re_viewer/src/open_url.rs:193]: nit

```suggestion

            DisplayMode::LocalRecordings => {
```
